src/Windows/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -105,7 +105,12 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
 
         public async Task<ExecutionResult> ResetAsync(bool initialize = true) {
             try {
+                string lastWorkingDirectory = null;
                 if (Session.IsHostRunning) {
+                    // Remember directory in case it is user-initiated reset 
+                    // so we can restore it when host completes the restart.
+                    lastWorkingDirectory = await Session.GetRWorkingDirectoryAsync();
```

Comment:
Do you want to preserve `lastWorkingDirectory` in case of an exception?

Reply:
Not sure, but why not?

Reply:
Ok